jubatus/util/concurrent/rwmutex.cpp

```diff
@@ -109,13 +145,23 @@ bool rw_mutex::impl::read_lock(double sec)
 bool rw_mutex::impl::write_lock()
 {
   if (!valid) return false;
+#ifdef JUBATUS_UTIL_CONCURRENT_RWMUTEX_ERRORCHECK
+  assert(pthread_rwlock_rdlock(&readers_lk)==0);
+  assert(readers.count(thread::id())==0);
```

Comment:
I think it is not necessary to check the conflict between write-lock and read-lock.
This function should check only the conflict between read-lock and read-lock.